This commit fixes the discontinuity when dealing with HWM/DWM. One ca…

rafaelluizmesquita opened the pull request:
…n now use 'off' (default), 'on', or 'smooth'. The code does that by fetching the Ap index from 3 days (the day in the datetime used in pyglow.Point, a day before and a day after), then performing a cubic interpolation to smooth out the Ap values (which are only calculated to every 3 hours).

Lastly, a script called 'tester.py' is placed for testing the validity of this procedure by plotting the 3-hour Ap, the smooth Ap, and HWM14 winds (with 'on', 'off', and 'smooth' flags) vs time.

A later commit should feature the Hp index (which is defined the same way as the Ap, but its calculation is at every 30 minutes - see https://www.gfz-potsdam.de/en/hpo-index for details).

rafaelluizmesquita committed:
This commit fixes the discontinuity when dealing with HWM/DWM. One can now use 'off' (default), 'on', or 'smooth'. The code does that by fetching the Ap index from 3 days (the day in the datetime used in pyglow.Point, a day before and a day after), then performing a cubic interpolation to smooth out the Ap values (which are only calculated to every 3 hours).

Lastly, a script called 'tester.py' is placed for testing the validity of this procedure by plotting the 3-hour Ap, the smooth Ap, and HWM14 winds (with 'on', 'off', and 'smooth' flags) vs time.

A later commit should feature the Hp index (which is defined the same way as the Ap, but its calculation is at every 30 minutes - see https://www.gfz-potsdam.de/en/hpo-index for details).

bharding512 commented:
Overall I think this is great, thanks! One detail, I think the default behavior should be 'on'. I don't know whether 'on' or 'off' makes more sense from first principles, but the current behavior is 'on'. It would be best if this update wouldn't change anybody's results.

Could you change it? Then I'll test it and merge if everything looks good.

rafaelluizmesquita committed:
This commit fixes the discontinuity when dealing with HWM/DWM. One can now use 'off' (default), 'on', or 'smooth'. The code does that by fetching the Ap index from 3 days (the day in the datetime used in pyglow.Point, a day before and a day after), then performing a cubic interpolation to smooth out the Ap values (which are only calculated to every 3 hours).

Lastly, a script called 'tester.py' is placed for testing the validity of this procedure by plotting the 3-hour Ap, the smooth Ap, and HWM14 winds (with 'on', 'off', and 'smooth' flags) vs time.

bharding512 commented:
Closed by PR#152
